irohad/ametsuchi/impl/postgres_block_index.cpp

```diff
@@ -45,29 +47,27 @@ namespace iroha {
           + ");");
     }
 
-    auto PostgresBlockIndex::indexAccountAssets(
-        const std::string &account_id,
-        const std::string &height,
-        const std::string &index,
-        const model::Transaction::CommandsType &commands) {
+    auto PostgresBlockIndex::indexAccountAssets(const std::string &account_id,
+                                                const std::string &height,
+                                                const std::string &index,
+                                                const CommandsType &commands) {
       // flat map abstract commands to transfers
       auto transfers =
-          commands | boost::adaptors::transformed([](const auto &cmd) {
-            return std::dynamic_pointer_cast<model::TransferAsset>(cmd);
-          })
-          | boost::adaptors::filtered(
-                [](const auto &cmd) { return bool(cmd); });
+          commands | boost::adaptors::filtered([](const auto &cmd) {
+            return cmd->get().which() == 15;
+          });
 
       return std::accumulate(
           transfers.begin(),
           transfers.end(),
           true,
           [&](auto &status, const auto &cmd) {
-            for (const auto &id : {cmd->src_account_id, cmd->dest_account_id}) {
+            auto command = boost::get<w<shared_model::interface::TransferAsset>>(cmd->get());
+            for (const auto &id : {command->srcAccountId(), command->destAccountId()}) {
```

Comment:
maybe just inline instead of loop:
```cpp
status &= this->indexAccountIdHeight(command->srcAccountId(), height)
    & this->indexAccountIdHeight(command->destAccountId(), height);
```
Imo much easier to understand

Reply:
I just replaced old member reference to the new one, but yeah, it looks much more readable